src/rpc/blockchain.cpp

```diff
@@ -455,23 +455,23 @@ static RPCHelpMan getdifficulty()
     RPCResult{RPCResult::Type::NUM, "height", "block height when transaction entered pool"},
     RPCResult{RPCResult::Type::NUM, "descendantcount", "number of in-mempool descendant transactions (including this one)"},
     RPCResult{RPCResult::Type::NUM, "descendantsize", "virtual transaction size of in-mempool descendants (including this one)"},
-    RPCResult{RPCResult::Type::STR_AMOUNT, "descendantfees", "modified fees (see above) of in-mempool descendants (including this one) (DEPRECATED)"},
+    RPCResult{RPCResult::Type::STR_AMOUNT, "descendantfees", "transaction fee rate of in-mempool descendants (including this one) in " + CURRENCY_ATOM + "/vB with fee rate deltas used for mining priority (DEPRECATED, returned only if config option -deprecatedrpc=fees is passed)"},
     RPCResult{RPCResult::Type::NUM, "ancestorcount", "number of in-mempool ancestor transactions (including this one)"},
     RPCResult{RPCResult::Type::NUM, "ancestorsize", "virtual transaction size of in-mempool ancestors (including this one)"},
-    RPCResult{RPCResult::Type::STR_AMOUNT, "ancestorfees", "modified fees (see above) of in-mempool ancestors (including this one) (DEPRECATED)"},
+    RPCResult{RPCResult::Type::STR_AMOUNT, "ancestorfees", "transaction fee rate of in-mempool ancestors (including this one) in " + CURRENCY_ATOM + "/vB with fee rate deltas used for mining priority (DEPRECATED, returned only if config option -deprecatedrpc=fees is passed)"},
     RPCResult{RPCResult::Type::STR_HEX, "wtxid", "hash of serialized transaction, including witness data"},
     RPCResult{RPCResult::Type::OBJ, "fees", "",
         {
-            RPCResult{RPCResult::Type::STR_AMOUNT, "base", "transaction fee in " + CURRENCY_UNIT},
-            RPCResult{RPCResult::Type::STR_AMOUNT, "modified", "transaction fee with fee deltas used for mining priority in " + CURRENCY_UNIT},
-            RPCResult{RPCResult::Type::STR_AMOUNT, "ancestor", "modified fees (see above) of in-mempool ancestors (including this one) in " + CURRENCY_UNIT},
-            RPCResult{RPCResult::Type::STR_AMOUNT, "descendant", "modified fees (see above) of in-mempool descendants (including this one) in " + CURRENCY_UNIT},
+            RPCResult{RPCResult::Type::STR_AMOUNT, "base", "transaction fee rate in " + CURRENCY_UNIT + "/kvB"},
+            RPCResult{RPCResult::Type::STR_AMOUNT, "modified", "modified fee rate in " + CURRENCY_UNIT + "/kvB with fee rate deltas used for mining priority"},
+            RPCResult{RPCResult::Type::STR_AMOUNT, "ancestor", "transaction fee rate of in-mempool ancestors (including this one) in " + CURRENCY_UNIT + "/kvB with fee rate deltas used for mining priority"},
+            RPCResult{RPCResult::Type::STR_AMOUNT, "descendant", "transaction fee rate of in-mempool descendants (including this one) in " + CURRENCY_UNIT + "/kvB with fee rate deltas used for mining priority"},
```

Comment:
`CTxMemPoolEntry.GetModifiedFee()`, `CTxMemPoolEntry.GetFee()`, `CTxMemPoolEntry.GetModFeesWithAncestors()`, `CTxMemPoolEntry.GetModFeesWithDescendants()` return `CAmount`s. See their [declarations](https://github.com/bitcoin/bitcoin/blob/3755dea3c502f32048b110a4868daefe9dbe3cc1/src/txmempool.h#L125-L154) in txmempool.h.

We aren't dividing these fee amounts by sizes, just pushing them directly into the result object in `entryToJSON`.

Reply:
Interesting. In `CTxMemPool::PrioritiseTransaction()`, is `TxMempoolInfo::nFeeDelta` incorrectly described as a fee rate? In BIP22 it's described as a fee. 
```cpp
LogPrintf("PrioritiseTransaction: %s feerate += %s\n", hash.ToString(), FormatMoney(nFeeDelta));
```

Reply:
you are correct! my apologies, i should have double-checked the declarations before making the change :grimacing: